examples/nlp/machine_translation/conf/aayn_base.yaml

```diff
@@ -0,0 +1,141 @@
+model:
+  beam_size: 4
+  len_pen: 0.6
+  max_generation_delta: 5
+  label_smoothing: 0.1
+
+  train_ds:
+    src_file_name: ???
+    tgt_file_name: ???
+    tokens_in_batch: 512
+    clean: true
+    max_seq_length: 512
+    cache_ids: true
+    cache_data_per_node: false
+    use_cache: true
+    shuffle: true
+    num_samples: -1
+    drop_last: false
+    pin_memory: false
+    num_workers: 8
+
+  validation_ds:
+    src_file_name: ???
+    tgt_file_name: ???
+    tokens_in_batch: 512
+    clean: false
+    max_seq_length: 512
+    cache_ids: true
+    cache_data_per_node: false
+    use_cache: true
+    shuffle: false
+    num_samples: -1
+    drop_last: false
+    pin_memory: false
+    num_workers: 8
+
+  test_ds:
+    src_file_name: ???
+    tgt_file_name: ???
+    tokens_in_batch: 512
+    clean: false
+    max_seq_length: 512
+    cache_ids: true
+    cache_data_per_node: false
+    use_cache: true
+    shuffle: false
+    num_samples: -1
+    drop_last: false
+    pin_memory: false
+    num_workers: 8
+
+  optim:
+    name: adam
+    lr: 0.001
+    betas:
+      - 0.9
+      - 0.98
+    weight_decay: 0.0
+    sched:
+      name: InverseSquareRootAnnealing
+      min_lr: 0.0
+      last_epoch: -1
+      warmup_ratio: 0.1
+
+  encoder_tokenizer:
+    tokenizer_name: yttm
+    tokenizer_model: ???
+    vocab_file: null
+    special_tokens: null
+
+  decoder_tokenizer:
+    tokenizer_name: yttm
+    tokenizer_model: ???
+    vocab_file: null
+    special_tokens: null
+
+  encoder_embedding:
+    max_sequence_length: 512
+    num_token_types: 2
+    embedding_dropout: 0.1
+    learn_positional_encodings: false
+    _target_: nemo.collections.nlp.modules.common.transformer.TransformerEmbedding
+
+  encoder:
+    hidden_size: 512
+    num_layers: 6
+    inner_size: 2048
+    num_attention_heads: 8
+    ffn_dropout: 0.1
+    attn_score_dropout: 0.1
+    attn_layer_dropout: 0.1
+    hidden_act: relu
+    mask_future: false
+    pre_ln: false
+    _target_: nemo.collections.nlp.modules.common.transformer.TransformerEncoder
+
+  decoder_embedding:
+    max_sequence_length: 512
+    num_token_types: 2
+    embedding_dropout: 0.1
+    learn_positional_encodings: false
+    _target_: nemo.collections.nlp.modules.common.transformer.TransformerEmbedding
+
+  decoder:
+    hidden_size: 512
+    inner_size: 2048
+    num_layers: 6
+    num_attention_heads: 8
+    ffn_dropout: 0.1
+    attn_score_dropout: 0.1
+    attn_layer_dropout: 0.1
+    hidden_act: relu
+    pre_ln: false
+    _target_: nemo.collections.nlp.modules.common.transformer.TransformerDecoder
+
+  head:
+    hidden_size: 512
+    num_classes: 37000
```

Comment:
This should be `decoder_vocab_size` but we don't have it anywhere in the config. I'm thinking that we should add 
```
decoder_embedding:
  vocab_size: 37000
```
And then interpolate this here:
```
  num_classes: ${model.decoder_embedding.vocab_size}
```